### PR TITLE
pkg/pdutil(ticdc): fix a deadlock issue

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close    https://github.com/pingcap/tiflow/issues/7201

### What is changed and how it works?
Fix a deadlock issue. Release the lock as soon as possible, otherwise `Run` and `Close` will cause a deadlock.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by gozssky — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

File: pkg/pdutil/clock.go [hunk 1]
```diff
@@ -120,8 +120,8 @@ func (c *clock) CurrentTime() (time.Time, error) {
 // Stop clock.
 func (c *clock) Stop() {
 	c.mu.Lock()
-	defer c.mu.Unlock()
 	c.cancel()
+	c.mu.Unlock()
```
Comment (amyangfei) [inline pkg/pdutil/clock.go:124]: Do we need to add a unit test

Comment (hi-rustin) [reply pkg/pdutil/clock.go:124]: Actually, this will fix that unstable test. So I don't think we need to add a new test.

**Pull request merged**